addons/account/models/account_payment.py

```diff
@@ -212,7 +214,11 @@ def _onchange_partner_id(self):
                 self.partner_bank_account_id = self.partner_id.commercial_partner_id.bank_ids[0]
             else:
                 self.partner_bank_account_id = False
-        return {'domain': {'partner_bank_account_id': [('partner_id', 'in', [self.partner_id.id, self.partner_id.commercial_partner_id.id])]}}
+        if self.payment_type == 'inbound' and self.company_id:
+            domain = {'partner_bank_account_id': [('partner_id', '=', self.company_id.partner_id.id)]}
```

Comment:
just to be more inclusive in case of a complex configuration 
```
domain = {'partner_bank_account_id': [('partner_id', 'in', [self.invoice_ids[0].company_id.partner_id.id, self.invoice_ids[0].company_id.partner_id.commercial_partner_id.id])]}

Reply:
Done